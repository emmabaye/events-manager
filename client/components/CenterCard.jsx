import React, { Component } from 'react';
import { Link } from 'react-router-dom';

/**
 * React component for center card
 * displayed for regular user.
 */
export default class CenterCard extends Component {
  /**
   * React's method to render react component.
   *
   * @return {object}
   */
  render() {
    return (
      <div className="col-md-4 event" style={{ height: "300px" }}>
        <div className="card">
          <div className="card-block" style={{ height: "300px" }}>
            <h5 className="card-title"><b>{this.props.centerDetails.name}</b></h5>
            <p className="card-text h6">{`${this.props.centerDetails.description.substr(0, 50)}...`}</p>
            <p className="card-text h6"><b>Location</b>:{`${this.props.centerDetails.location.substr(0, 25)}...`}</p>
            <p className="card-text h6"><b>Capacity</b>: {this.props.centerDetails.capacity} seats</p>
            <p className="card-text h6"><b>Price</b>: &#8358;{this.props.centerDetails.price}</p>
            <p className="card-text">
              <b>Availability</b>: {
                (this.props.centerDetails.available === 'true') ? "Center is Available" : "Not Available"
              }
            </p>

            <p className="text-center">
              <Link to={`/centers/${this.props.centerDetails.id}`} className="btn btn-sm btn-primary">
                <i className="fa fa-info-circle fa-lg" /> View Center
              </Link>
            </p>
          </div>
        </div>
      </div>
    );
  }
}
