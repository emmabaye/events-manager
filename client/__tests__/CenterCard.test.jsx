import { BrowserRouter } from 'react-router-dom';
import CenterCard from '../components/CenterCard.jsx';

describe('CenterCard Component', () => {
  const props = {
    centerDetails: {
      id: "1",
      name: "",
      description: {
        substr: () => {}
      },
      location: {
        substr: () => {}
      },
      capacity: "",
      price: "",
      available: 'true'
    },
    show: () => {}
  };

  it('it should render CenterCard component', () => {
    const wrapper = shallow(<BrowserRouter><CenterCard {...props} /></BrowserRouter>);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render CenterCard component', () => {
    props.centerDetails.available = 'false';
    const wrapper = mount(<BrowserRouter><CenterCard {...props} /></BrowserRouter>);
    expect(wrapper.length).toEqual(1);
  });

});
