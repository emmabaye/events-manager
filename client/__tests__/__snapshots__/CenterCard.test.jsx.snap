// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterCard Component it should render CenterCard component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <CenterCard
    centerDetails={
      Object {
        "available": "true",
        "capacity": "",
        "description": Object {
          "substr": [Function],
        },
        "id": "1",
        "location": Object {
          "substr": [Function],
        },
        "name": "",
        "price": "",
      }
    }
    show={[Function]}
  />
</Router>
`;
